In [1]:
gpu = True
# gpu = False
if gpu:
    from tools.utils import enableGPU, allow_growth
    gpu_id = 0
    enableGPU(gpu_id)
    allow_growth()
else:
    from tools.utils import disableGPU
    disableGPU()
    
    
import warnings
warnings.filterwarnings("ignore", "", RuntimeWarning)
warnings.filterwarnings("ignore", "", FutureWarning)
from validation import calc_val_metrics2, np

In [2]:
model_name = 'digits_epoch-74_loss-0.1696_acc-0.9559.h5'
n = None # 7100, 5100

In [3]:
version = "v7"
# img_list_filename = f"schwarzkopf-price-tags-val-imgs"
# boxes_arrays_filename = f"schwarzkopf-price-tags-val-arrays"
# suffix = 'train'
suffix = 'train_2'
# suffix = 'val_2'

boxes_arrays_filename = f"union_companies_labels_{suffix}_det_{suffix}"
img_list_filename = f"union_companies_imgs_{suffix}_det_{suffix}"
max_side = 120
gt_thresh = .2
divide_algo = True

In [4]:
version = "v9"
suffix = 'train_2'

boxes_arrays_filename = f"union_companies_labels_{suffix}_det_{suffix}"
img_list_filename = f"union_companies_imgs_{suffix}_det_{suffix}"

metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2024/2024 [04:30<00:00,  7.47it/s]


acc_micro: 65.2, acc_macro: 70.5, f1_micro: 79.0, f1_macro: 80.0, pr_macro: 88.2, rec_macro: 79.4


In [4]:
version = "v10"
suffix = 'train_2'
boxes_arrays_filename = f"union_companies_labels_{suffix}_det_{suffix}"
img_list_filename = f"union_companies_imgs_{suffix}_det_{suffix}"

metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2024/2024 [07:47<00:00,  4.05it/s]


acc_micro: 65.5, acc_macro: 70.8, f1_micro: 79.1, f1_macro: 80.4, pr_macro: 84.9, rec_macro: 82.4


In [7]:
version = "v9"
suffix = 'train'
boxes_arrays_filename = f"union_companies_labels_{suffix}_det_{suffix}"
img_list_filename = f"union_companies_imgs_{suffix}_det_{suffix}"

metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2021/2021 [05:28<00:00,  6.15it/s]


acc_micro: 72.0, acc_macro: 78.1, f1_micro: 83.7, f1_macro: 85.2, pr_macro: 86.3, rec_macro: 90.0


In [11]:
version = "v9"
suffix = 'test'
boxes_arrays_filename = f"union_companies_labels_{suffix}"
img_list_filename = f"union_companies_imgs_{suffix}"

metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2024/2024 [08:39<00:00,  4.16it/s]


acc_micro: 70.7, acc_macro: 77.0, f1_micro: 82.8, f1_macro: 84.4, pr_macro: 86.1, rec_macro: 89.0


In [12]:
version = "v10"
suffix = 'test_2'
boxes_arrays_filename = f"union_companies_labels_{suffix}"
img_list_filename = f"union_companies_imgs_{suffix}"


metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2025/2025 [10:57<00:00,  2.95it/s]


acc_micro: 65.5, acc_macro: 71.1, f1_micro: 79.2, f1_macro: 80.8, pr_macro: 85.9, rec_macro: 81.9


In [27]:
version = "v9"
metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2021/2021 [04:40<00:00,  6.77it/s]


acc_micro: 61.6, acc_macro: 70.1, f1_micro: 76.2, f1_macro: 79.7, pr_macro: 74.6, rec_macro: 93.3


In [28]:
version = "v6"
metrics = calc_val_metrics2(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 2021/2021 [09:52<00:00,  3.53it/s]


acc_micro: 62.3, acc_macro: 70.3, f1_micro: 76.8, f1_macro: 79.7, pr_macro: 75.0, rec_macro: 92.6


In [3]:
version = "v6"
# version = "v8"
max_side = 120
img_list_filename = f"schwarzkopf-price-tags-val-imgs-size-{max_side}"
boxes_arrays_filename = f"schwarzkopf-price-tags-val-arrays-size-{max_side}"
gt_thresh = .2
divide_algo = True

In [4]:
metrics = calc_val_metrics(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

100%|██████████| 5167/5167 [44:14<00:00,  1.99it/s] 


acc_micro: 60.6, acc_macro: 63.3, f1_micro: 75.5, f1_macro: 70.7, pr_macro: 94.4, rec_macro: 66.3


In [3]:
version = "v6"
max_side = 120
img_list_filename = f"schwarzkopf-price-tags-val-imgs-size-{max_side}"
boxes_arrays_filename = f"schwarzkopf-price-tags-val-arrays-size-{max_side}"
gt_thresh = .2
divide_algo = True

In [3]:
metrics = calc_val_metrics(version, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side, box_expand_size)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/keras/layers/core.py:986: UserWarning: models.digits_classifier is not loaded, but a Lambda layer uses it. It may cause errors.
  , UserWarning)
100%|██████████| 5167/5167 [52:46<00:00,  1.52it/s]  


acc_micro: 41.5, acc_macro: 48.3, f1_micro: 58.6, f1_macro: 59.3, pr_macro: 72.1, rec_macro: 61.1


In [ ]:
dataset_name = "v1"
max_side = 120
img_list_filename = f"schwarzkopf-price-tags-train-imgs-size-{max_side}"
boxes_arrays_filename = f"schwarzkopf-price-tags-train-arrays-size-{max_side}"
gt_thresh = .2
divide_algo = True
metrics = calc_val_metrics(dataset_name, model_name, img_list_filename,
                           boxes_arrays_filename, n, gt_thresh, divide_algo, max_side)
print("acc_micro: %.1f, acc_macro: %.1f, f1_micro: %.1f, f1_macro: %.1f, pr_macro: %.1f, rec_macro: %.1f" % tuple([x * 100 for x in metrics]))

**Val**  
Default v3. old params v1.  
acc_micro: 59.3, acc_macro: 61.8, f1_micro: 74.4, f1_macro: 69.1, pr_macro: 94.8, rec_macro: 64.6

Default v3. params v1. expand size=2   
acc_micro: 59.7, acc_macro: 62.4, f1_micro: 74.7, f1_macro: 70.1, pr_macro: 92.5, rec_macro: 66.4

Default v3. params v1. expand size=3  
acc_micro: 59.6, acc_macro: 62.4, f1_micro: 74.7, f1_macro: 70.2, pr_macro: 92.7, rec_macro: 66.1

Default v3. new params v2(recall)   
acc_micro: 60.2, acc_macro: 63.1, f1_micro: 75.2, f1_macro: 70.6, pr_macro: 93.8, rec_macro: 66.5  

Default v3. New params v3.(acc)  
acc_micro: 60.5, acc_macro: 63.1, f1_micro: 75.4, f1_macro: 70.5, pr_macro: 94.3, rec_macro: 66.2  

Default v3. New params v4.(acc)   
acc_micro: 60.7, acc_macro: 63.4, f1_micro: 75.5, f1_macro: 70.8, pr_macro: 94.3, rec_macro: 66.4   

Default v3. New params v6.(acc).expsize=1    
acc_micro: 60.6, acc_macro: 63.3, f1_micro: 75.5, f1_macro: 70.8, pr_macro: 94.4, rec_macro: 66.4

Default v4(kind=stable) New params v6.(acc).expsize=1    
acc_micro: 60.6, acc_macro: 63.3, f1_micro: 75.5, f1_macro: 70.7, pr_macro: 94.4, rec_macro: 66.3

Default v3. New params v7.(acc)   
acc_micro: 61.1, acc_macro: 64.2, f1_micro: 75.8, f1_macro: 72.0, pr_macro: 92.1, rec_macro: 68.5

v6_2  
acc_micro: 60.8, acc_macro: 63.4, f1_micro: 75.6, f1_macro: 70.8, pr_macro: 94.5, rec_macro: 66.3  

v7. algo v2  
acc_micro: 58.7, acc_macro: 61.7, f1_micro: 74.0, f1_macro: 70.1, pr_macro: 91.4, rec_macro: 66.3

v8.  
acc_micro: 61.7, acc_macro: 64.6, f1_micro: 76.3, f1_macro: 72.1, pr_macro: 93.1, rec_macro: 68.2  

v2  
Default.  
acc_micro: 61.6, acc_macro: 64.1, f1_micro: 76.2, f1_macro: 71.4, pr_macro: 94.8, rec_macro: 66.8, speed: 4.57it/s  
                        
mser+swt   
acc_micro: 62.1, acc_macro: 64.5, f1_micro: 76.6, f1_macro: 71.9, pr_macro: 94.4, rec_macro: 67.5  3.18it/s (медленнее на 43%)


mser+swt. optimized params.  
acc_micro: 63.0, acc_macro: 65.4, f1_micro: 77.3, f1_macro: 72.6, pr_macro: 94.2, rec_macro: 68.4  2.13it/s 3.49it/s

switched filters, optimized threshold  
acc_micro: 61.5, acc_macro: 64.0, f1_micro: 76.2, f1_macro: 71.3, pr_macro: 94.8, rec_macro: 66.7 speed 6.04it/s  

switched filters, not optimized threshold   
acc_micro: 60.5, acc_macro: 62.8, f1_micro: 75.4, f1_macro: 70.3, pr_macro: 94.9, rec_macro: 65.4  

swt only  
acc_micro: 47.8, acc_macro: 49.3, f1_micro: 64.7, f1_macro: 57.2, pr_macro: 96.8, rec_macro: 50.4 speed 7.79it/s

Train

mser+swt  
acc_micro: 63.3, acc_macro: 66.4, f1_micro: 77.5, f1_macro: 73.6, pr_macro: 94.9, rec_macro: 69.1 speed 3.84it/s   4.24it/s

mser  
acc_micro: 62.9, acc_macro: 65.9, f1_micro: 77.2, f1_macro: 73.1, pr_macro: 95.3, rec_macro: 68.4 speed  5.01it/s 4.42it/s


Algo Ver3. prev params  
acc_micro: 60.7, acc_macro: 63.7, f1_micro: 75.6, f1_macro: 70.8, pr_macro: 95.2, rec_macro: 66.1  

Algo Ver3. New params: acc macro optimiziation  
acc_micro: 61.8, acc_macro: 65.0, f1_micro: 76.4, f1_macro: 72.3, pr_macro: 94.9, rec_macro: 67.8  

On Validation. Hyperparam search.  

**Schwarzkopf**           
gt .2
acc_micro: 60.6, acc_macro: 63.1, f1_micro: 75.5, f1_macro: 70.4  

**New space, fixed param bug**    
acc_micro: 61.0, acc_macro: 63.4, f1_micro: 75.8, f1_macro: 70.6  

**New params. More steps**  
acc_micro: 61.0, acc_macro: 63.5, f1_micro: 75.8, f1_macro: 70.8  

**New params. More steps2**  
acc_micro: 61.2, acc_macro: 63.6, f1_micro: 75.9, f1_macro: 71.0

**New params. More steps (plus new algo v2)**  
acc_micro: 61.2, acc_macro: 63.8, f1_micro: 75.9, f1_macro: 71.1  

**New params. More steps (plus new algo v3)**  
acc_micro: 61.2, acc_macro: 63.9, f1_micro: 76.0, f1_macro: 71.1  

**New params. More steps (plus new algo v4)**  
acc_micro: 61.2, acc_macro: 63.8, f1_micro: 75.9, f1_macro: 71.1  

**New params. More steps2 (algo ver4)**  
acc_micro: 61.6, acc_macro: 64.0, f1_micro: 76.2, f1_macro: 71.4  

**Same. first 2 filters switched**  
acc_micro: 61.6, acc_macro: 64.1, f1_micro: 76.2, f1_macro: 71.4


64.0 - 58.9=5.1

On Validation. Default params.  
**Schwarzkopf**    
**Changed metrics**  
**Before resize issue with train**  
acc_micro: 55.2, acc_macro: 57.5, f1_micro: 71.2, f1_macro: 64.6  

**After fixing resize issue(same scale x/y)**  
acc_micro: 55.8, acc_macro: 58.0, f1_micro: 71.6, f1_macro: 65.0

diff scale x/y  (gt_thresh .5)  
acc_micro: 56.6, acc_macro: 58.7, f1_micro: 72.3, f1_macro: 65.6  
gt_thresh .4   
acc_micro: 56.9, acc_macro: 58.8, f1_micro: 72.5, f1_macro: 65.8  
gt_thresh .6  
acc_micro: 56.9, acc_macro: 58.8, f1_micro: 72.5, f1_macro: 65.8

gt_thresh .2  
acc_micro: 57.0, acc_macro: 58.9, f1_micro: 72.6, f1_macro: 65.9  

**Schwarzkopf**  
***Hyperparams***    
gt_thresh .2  
acc_micro: 61.3, acc_macro: 64.6, f1_micro: 76.0, f1_macro: 71.9   

**New space, fixed param bug**  
acc_micro: 61.9, acc_macro: 64.9, f1_micro: 76.5, f1_macro: 72.1   

**Find params. More steps**  
acc_micro: 62.0, acc_macro: 65.2, f1_micro: 76.6, f1_macro: 72.4

**new algo version1. new params**  
acc_micro: 62.1, acc_macro: 65.2, f1_micro: 76.7, f1_macro: 72.4  

**New algo version2. same params**  
acc_micro: 62.2, acc_macro: 65.3, f1_micro: 76.7, f1_macro: 72.5  

**New algo version3. same params**  
acc_micro: 62.4, acc_macro: 65.4, f1_micro: 76.8, f1_macro: 72.6  

**New algo version4. same params**  
acc_micro: 62.4, acc_macro: 65.4, f1_micro: 76.8, f1_macro: 72.6  

**New params2. No divide algo**  
acc_micro: 62.5, acc_macro: 65.6, f1_micro: 76.9, f1_macro: 72.8  


**New params2. New algo version4. Max side 120**  
acc_micro: 62.9, acc_macro: 65.9, f1_micro: 77.2, f1_macro: 73.1  

**New params2. New algo version4. Max side 200**   
acc_micro: 63.9, acc_macro: 66.8, f1_micro: 78.0, f1_macro: 73.9    

**New params3. Divide algo optimized**  
acc_micro: 63.0, acc_macro: 66.1, f1_micro: 77.3, f1_macro: 73.2

***Default algo params***  
**Default area, ar params(same scale x/y)**  
acc_micro: 56.6, acc_macro: 59.7, f1_micro: 72.3, f1_macro: 66.7

**Fix scaling (diff scale x/y)**   
gt_thresh .5  
acc_micro: 57.3, acc_macro: 60.2, f1_micro: 72.8, f1_macro: 67.2  

gt_thresh .4  
acc_micro: 57.9, acc_macro: 60.6, f1_micro: 73.4, f1_macro: 67.6

gt_thresh .3  
acc_micro: 58.1, acc_macro: 60.7, f1_micro: 73.5, f1_macro: 67.7  

gt_thresh .2  
acc_micro: 58.2, acc_macro: 60.7, f1_micro: 73.5, f1_macro: 67.7 

gt_thresh .1  
acc_micro: 58.2, acc_macro: 60.7, f1_micro: 73.6, f1_macro: 67.7  

**grad pruning**  
чекаем что угол не сильно отличается.  
acc_micro: 62.6, acc_macro: 65.6, f1_micro: 77.0, f1_macro: 72.8  

check that edge intensity is high enoough before deleting point (14 минут инференс)   
acc_micro: 62.2, acc_macro: 65.1, f1_micro: 76.7, f1_macro: 72.4  

no preprocessing blur. step in direction of angle from edge and delete point from region, max_len 3    
acc_micro: 62.0, acc_macro: 64.9, f1_micro: 76.5, f1_macro: 72.2

preprocessing median blur. step in direction of angle from edge and delete point from region, max_len 3    
acc_micro: 62.1, acc_macro: 65.1, f1_micro: 76.6, f1_macro: 72.3

gaussian blur before gradients  
acc_micro: 62.6, acc_macro: 65.7, f1_micro: 77.0, f1_macro: 72.9  

median blur before gradients  
acc_micro: 62.7, acc_macro: 65.7, f1_micro: 77.1, f1_macro: 73.0

threshold - middle between median and max  
acc_micro: 62.6, acc_macro: 65.6, f1_micro: 77.0, f1_macro: 72.8  

threshold - median  
acc_micro: 62.0, acc_macro: 64.9, f1_micro: 76.5, f1_macro: 72.0  

default  
acc_micro: 62.9, acc_macro: 65.9, f1_micro: 77.2, f1_macro: 73.1  

On Train.   
**Mars**  
**Default params**  
acc_micro: 58.7, acc_macro: 59.7, f1_micro: 74.0, f1_macro: 65.8


On Validation. Default params.  
**Mars**  
acc_micro: 59.0, acc_macro: 59.7, f1_micro: 74.2, f1_macro: 65.9